### DISCUSSION TITLE
Why is eslint-define-config not picking up @eslint-types/typescript-eslint errors?

### DISCUSSION THREAD

**Discussion by U-4-E-A** [Shinigami92/eslint-define-config#266] [General]
Firstly, I have `@eslint-types/typescript-eslint` installed.

I have it set up exactly as per the example here: -

[https://www.npmjs.com/package/eslint-define-config#usage](https://www.npmjs.com/package/eslint-define-config#usage)

```
// @ts-check
const { defineConfig } = require('eslint-define-config')

/// <reference types="@eslint-types/typescript-eslint" />

module.exports = defineConfig({
  parserOptions: {
    project: './tsconfig.json'
  },
  parser: '@typescript-eslint/parser',
  plugins: ['@typescript-eslint'],
  rules: {
    '@typescript-eslint/adjacent-overload-signatures': ['error']
  }
})
```
However, if I put a deliberate error (typo) in the rule name ("@typescript-eslint/adjacent-1234567890") I do not get an error. The error does however work for the standard eslint types - it is specifically the @eslint-types/typescript-eslint errors that are not showing.

What am I doing wrong? Thanks.

**Comment by Shinigami92** [comment]
with the context you given, everything should work fine, assuming you also installed everything needed
would you like to setup a minimal reproducible repo/stackblitz, so I could investigate what's missing?

**Reply by U-4-E-A** [reply]
I started playing around with eslint-define-config last night and I couldn't initially get it working with the basic package (thankfully I did as it's an amazing package). I took a copy of this sandbox I randomly found so I had a working repo: -

https://codesandbox.io/p/devbox/template-7-0-forked-xws04c?file=%2Fpackage.json

I then figured out it was because my vscode settings had this in it: -

```
{
    "javascript.validate.enable": false
}
```

I changed that to "true" and it started picking up the type errors. I then installed `@eslint-types/typescript-eslint` to my copy of the repo and added `/// <reference types="@eslint-types/typescript-eslint" />` (as seen above) but it won't type check the `@typescript-eslint` rules. I've restarted vscode and still no luck.

I wonder if this is something to do with my vscode settings?

**Reply by Shinigami92** [reply]
So in the package.json from above's link, I cannot see `@eslint-types/typescript-eslint` in the `devDependencies` :thinking: 
You know that you need to install that?
e.g. `yarn add -D @eslint-types/typescript-eslint`
but this is also documented in the README: https://github.com/eslint-types/eslint-define-config#usage

**Reply by U-4-E-A** [reply]
Sorry, I should have mentioned that I did add that and it is still not working.

OK, so I think I am zeroing in on the issue. If I use `v1.24.1` of the package with this installation, it works just fine: -

```
{
  "name": "template-70",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "eslint-define-config": "^1.24.1"
  },
  "devDependencies": {
    "@eslint-types/typescript-eslint": "^6.12.0",
    "npm-upgrade": "^3.1.0"
  }
}
```
However, if I upgrade to `v2.0.0`, it stops working: -

```
{
  "name": "template-70",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "eslint-define-config": "^2.0.0"
  },
  "devDependencies": {
    "@eslint-types/typescript-eslint": "^6.12.0",
    "npm-upgrade": "^3.1.0"
  }
}
```

Is this a bug or am I doing something wrong here?

**Reply by Shinigami92** [reply]
v2 did introduce that new feature, that's why it is a major release with breaking changes
I need to investigate later on, but for now: enjoy the holidays

**Reply by U-4-E-A** [reply]
Thanks. You too!

**Reply by Shinigami92** [reply]
It's very weird, I tried also to setup a stackblitz and had not luck.
But when I checkout https://github.com/faker-js/faker and use VSCode locally, everything works fine :shrug: 

Maybe it is the `"eslint.validate": ["javascript", "typescript"]` setting?

**Reply by U-4-E-A** [reply]
OK, through trial and error, I have found the culprit in my vcode settings: -

```
}
    typescript.tsserver.experimental.enableProjectDiagnostics: true
}
```
If that is set to `true`, it breaks `v2.0.0` of the package. If it is set to `false`, it works just fine.

A bit more about this setting can be found here: -

https://stackoverflow.com/questions/55201424/how-to-get-vscode-to-show-typescript-errors-for-files-not-open-in-the-editor

Is this considered a bug in the `eslint-define-config` package or do I have to change my settings?

**Reply by Shinigami92** [reply]
I would say, because it works without that specific setting just fine, it's not a bug with `eslint-define-config`
especially when it is an "experimental" setting

**Reply by U-4-E-A** [reply]
Do you have any idea exactly why it is happening? Presumably there was a change between `v1.24.1` and `v2.0.0` that caused that issue to show up?

**Reply by Shinigami92** [reply]
Absolutely no
Maybe ask a question in VSCode repo or TypeScript repo, as you identified an issue with that experimental setting

**Reply by U-4-E-A** [reply]
I'll do that and get back to you if I get something concrete as to why it is happening.

For now - for convenience sake - I am sticking with `v1.24.1`, which seems to be doing everything I want for now.
